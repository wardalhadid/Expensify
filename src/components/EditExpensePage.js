import React from "react";
import { connect} from "react-redux";
import { useNavigate, useParams } from "react-router";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";

const EditExpensePage = (props) => {
  let navigate = useNavigate();
  let {id} = useParams();
  const selectedExpense = props.expenses.find((expense) => expense.id === id);
  return (
   <div>
    <ExpenseForm
    expense={selectedExpense}
     onSubmit = {(expense) => {
     props.dispatch(editExpense(id, expense));
      navigate('/');
     }}
    />
    <button
     onClick={() => {
      props.dispatch(removeExpense({id}));
      navigate('/');
      }}>
       Remove
    </button>
    </div>
  );
}

const mapStateToProps = (state) => {
 return {
  expenses: state.expenses
 };
};


export default connect(mapStateToProps)(EditExpensePage);