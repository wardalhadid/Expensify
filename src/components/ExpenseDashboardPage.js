import React from "react";
import ConnectedExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashboardPage = () => {
return (
 <div>
   <ExpenseListFilters />
  <ConnectedExpenseList />
 </div>
);
}

export default ExpenseDashboardPage;