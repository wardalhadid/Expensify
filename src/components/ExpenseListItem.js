import React from "react";
import { Link } from "react-router-dom";
import numeral from 'numeraljs';
import moment from "moment";

const ExpenseListItem = ({description, amount, createdAt, id}) =>{
return (
  <div>
    <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
    </Link>
        <p>
          {numeral(amount).format('$0,0.00')}
           - 
           {moment(createdAt).format('MMMM Do, YYYY')}
           </p>
  </div>
);
}

export default ExpenseListItem;