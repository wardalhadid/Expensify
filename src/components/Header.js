import React from "react";
import {NavLink} from 'react-router-dom';


const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" className={({isActive}) => (isActive ? "is-active" : "")}>
     Dashboard 
    </NavLink>
    <NavLink to="/create" className={({isActive}) => (isActive ? "is-active" : "")}> 
     Create Expense 
    </NavLink>
    {/* <NavLink to="help" className={({isActive}) => (isActive ? "is-active" : "")}> 
     Help 
    </NavLink> */}
  </header>
);

export default Header;