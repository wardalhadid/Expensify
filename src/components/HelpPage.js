import React from "react";

const HelpPage = () => (
  <p>this is help page</p>
);

export default HelpPage;