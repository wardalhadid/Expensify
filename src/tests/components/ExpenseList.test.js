import {ExpenseList} from "../../components/ExpenseList";
import React from "react";
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import toJSON from 'enzyme-to-json';

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should render expense list with empty message', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

