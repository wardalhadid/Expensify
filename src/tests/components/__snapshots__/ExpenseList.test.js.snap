// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="credit card"
    id="3"
    key="3"
  />
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
